### ISSUE TITLE
Support GitHub deploy keys

### ISSUE THREAD

**Issue by dmolesUC3** [Apicurio/apicurio-studio#193]
The blanket permission Apicurio Studio asks for when integrating with GitHub is kind of bananas. I actually went so far as to create a brand-new GitHub account under a different email address in order to play around with the studio, because I didn't feel comfortable giving write access to all my public repositories at once, let alone read/write access to my organizations' private repositories. Adding support for [deploy keys](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys) would allow developers to give the studio access on a project-by-project basis.

**Comment by EricWittmann** [comment]
Yeah the granularity of the OAuth permissions supported by GH is not ideal.  You can grant/revoke access at the Org level, which is good, but it sure would be nice to only allow "write" access to a single repository.  I'll have a look at deploy keys and see if it can be easily used.  

Note that GitLab and Bitbucket aren't any better in this regard. :(

The good news is that Beta 2 of Apicurio will not require linking to GH, GL, or BB unless you want to *publish* an API Design to one of those platforms.  Instead, the content of the design will be stored in the Apicurio database.  And there will be a separate "Publish" on-demand action that a user can execute if she wants to publish to a source control system.

**Comment by EricWittmann** [comment]
So my initial reading of deploy keys indicates that they are useful if using git (just access the repository using a git client via SSH) but not if using the GH API (which is what Apicurio does).  The OAuth approach is specifically designed to grant a server access to GH data on behalf of a user.  Another approach is a GitHub app, which can be limited to certain repositories but has some drawbacks (e.g. the App must be installed on an organization by an org owner, and actions taken by the server do not reflect the user who performed them, but instead the App).

In any case, I do realize that this is a problem, and hope that the approach coming in Beta 2 helps.  If users are not comfortable linking their account, there will be a simple way to download the API Design and manually commit it to a repo (or obviously do whatever else).
